File: Ahorn/triggers/lightningColorTrigger.jl
module FrostHelperLightning

using ..Ahorn, Maple

@mapdef Trigger "FrostHelper/LightningColorTrigger" FHLightningColorTrigger(x::Integer, y::Integer, width::Integer=16, height::Integer=16, color1::String="fcf579", color2::String="8cf7e2", fillColor::String="ffffff", fillColorMultiplier::Number=0.1, persistent::Bool=true)

const placements = Ahorn.PlacementDict(
	"Lightning Color (Frost Helper)" => Ahorn.EntityPlacement(
		FHLightningColorTrigger,
		"rectangle"
	)
)

end

File: Loenn/triggers/lightningColorTrigger.lua
local jautils = require("mods").requireFromPlugin("libraries/jautils")

local lightningColorTrigger = {}
lightningColorTrigger.name = "FrostHelper/LightningColorTrigger"
lightningColorTrigger.category = "visual"

jautils.createPlacementsPreserveOrder(lightningColorTrigger, "normal", {
    { "width", 16 },
    { "height", 16 },
    { "color1", "fcf579", "color" },
    { "color2", "8cf7e2", "color" },
    { "fillColor", "ffffff", "color" },
    { "fillColorMultiplier", 0.1 },
    { "depth", -1000100, "depth" },
    { "targetGroup", "" },
    { "persistent", true },
})

return lightningColorTrigger